### ISSUE TITLE
Suggest using "await using" when possible

### ISSUE THREAD

**Issue by sylveon** [dotnet/roslyn#51418]
**Brief description:**

Currently, when a type that can be asynchronously disposed is synchronously disposed in an async method, no warning is emitted. A warning here would be helpful, and ReSharper already has this feature

**Languages applicable:**

C# only, as `await using` doesn't seem to be a thing in VB

**Code example that the analyzer should report:**

```cs
public static async Task Main()
{
    // should warn 'prefer "await using"'
    using var conn = new SqlConnection("");

    // should also warn
    using (var conn2 = new SqlConnection("")) { }
}
```

**Documentation requirements:**

When this analyzer is implemented, it must be documented by following the steps at [Documentation for IDE CodeStyle analyzers](https://github.com/dotnet/roslyn/blob/master/docs/contributing/Documentation%20for%20IDE%20CodeStyle%20analyzers.md).

**Comment by CyrusNajmabadi** [comment]
Moving to Roslyn-Analyzers.
